docs/user/howto_gateway_api.md

```diff
@@ -0,0 +1,411 @@
+remoteistios                                            sailoperator.io/v1alpha1            false        RemoteIstio
+authorizationpolicies               ap                  security.istio.io/v1                true         AuthorizationPolicy
+peerauthentications                 pa                  security.istio.io/v1                true         PeerAuthentication
+requestauthentications              ra                  security.istio.io/v1                true         RequestAuthentication
+telemetries                         telemetry           telemetry.istio.io/v1               true         Telemetry
+
+# A gateway class for applications to use gateways on
+$ oc get gatewayclass openshift-gateway-api -o yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  creationTimestamp: "2024-11-05T13:15:50Z"
+  generation: 1
+  name: openshift-gateway-api
+  resourceVersion: "1518"
+  uid: 2490e78e-9000-42cf-8234-d3e5482825a9
+spec:
+  controllerName: openshift.io/gateway-controller
+status:
+  conditions:
+  - lastTransitionTime: "2024-11-05T13:18:02Z"
+    message: Handled by Istio controller
+    observedGeneration: 1
+    reason: Accepted
+    status: "True"
+    type: Accepted
+
+# Service mesh operator and istio controller
+$ oc get deployment -n openshift-gateway-api
+NAME                           READY   UP-TO-DATE   AVAILABLE   AGE
+istiod-openshift-gateway-api   1/1     1            1           26h
+servicemesh-operator3          1/1     1            1           26h
+```
+These resources are handled by MicroShift and will always be present.
+
+## Configuring Gateways
+In order to use the feature we only need an application that we want to expose through a gateway, a gateway, and some routes.
+
+First we create a sample application:
+```bash
+$ oc create -f microshift/test/assets/hello-microshift.yaml 
+pod/hello-microshift created
+
+$ oc create -f microshift/test/assets/hello-microshift-service.yaml 
+service/hello-microshift created
+
+$ oc get pod,svc hello-microshift
+NAME                   READY   STATUS    RESTARTS   AGE
+pod/hello-microshift   1/1     Running   0          48s
+
+NAME                       TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+service/hello-microshift   ClusterIP   10.43.216.183   <none>        8080/TCP   13s
+```
+
+Then we can create a gateway for it:
+```bash
+$ cat gatewayapi/gateway.yaml 
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: demo-gateway
+  namespace: openshift-gateway-api
+spec:
+  gatewayClassName: openshift-gateway-api
+  listeners:
+  - name: demo
+    hostname: "*.microshift-9"
+    port: 8080
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: All
+
+$ oc create -f gatewayapi/gateway.yaml 
+gateway.gateway.networking.k8s.io/demo-gateway created
+
+$ oc get gateway -n openshift-gateway-api demo-gateway -o yaml | yq '.status'
```

Comment:
Can we further filter out the output to make it shorter? Probably the address part is the most interesting one, right?